test/rgb_matrix/animation/config/field_type/integer_test.exs

```diff
@@ -0,0 +1,67 @@
+defmodule IntegerTest do
+  use ExUnit.Case
+
+  alias RGBMatrix.Animation.Config.FieldType.Integer
+
+  defp test_integer(_context) do
+    [
+      test_integer: %Integer{
+        default: 5,
+        min: 1,
+        max: 11,
+        step: 2
+      }
+    ]
+  end
+
+  describe "Integer field type function `validate/2`" do
+    setup :test_integer
+
+    test "receives valid integer input and returns `:ok`", context do
+      assert Integer.validate(context.test_integer, 3) == :ok
+    end
+
+    test "receives invalid step-multiple input and returns `:validation_error`", context do
+      assert Integer.validate(context.test_integer, 4) == :validation_error
+    end
+
+    test "receives invalid out-of-range input and returns `:validation_error`", context do
+      assert Integer.validate(context.test_integer, 13) == :validation_error
+    end
+  end
+
+  describe "Integer field type function `cast/2`" do
+    setup :test_integer
+
+    test "receives valid string input and returns `{:ok, <value>}`", context do
+      assert Integer.cast(context.test_integer, "7") == {:ok, 7}
+    end
+
+    test "receives valid integer input and returns `{:ok, <value}`", context do
+      assert Integer.cast(context.test_integer, 7) == {:ok, 7}
+    end
+
+    test "receives float string input and returns the integer part only", context do
+      assert Integer.cast(context.test_integer, "1.0") == {:ok, 1}
+    end
+
+    test "receives float input and returns the integer part only", context do
+      assert Integer.cast(context.test_integer, 9.0) == {:ok, 9}
```

Comment:
I'm not sure I understand the question. Do you mean something in the range `0<x<1`? It would return `0`. If you mean something like `9.2`, it would return `9`

I wrote the test to cover the code, so I'm not 100% sure about the design intent of casting floats to integers. I should probably add a case, though.

Reply:
>If you mean something like `9.2`, it would return `9`

Exactly. I don't see a test for that and was unsure of the behavior.

>I'm not 100% sure about the design intent of casting floats to integers.

If in doubt, check with @doughsay. I'm not sure about what rounding assumptions we want to make.